py/visdom/__init__.py

```diff
@@ -1322,6 +1320,10 @@ def scatter(self, X, Y=None, win=None, env=None, opts=None, update=None,
 
                 data.append(_scrub_dict(_data))
 
+        if len(data) > 1:
```

Comment:
Technically this could actually be determined the moment we set `K`, as `K` is the number of lines.

Reply:
This is a bit awkward, because `K` is computed as `int(Y.max())`. So technically it is the max label index. However, we also seemed to have assumed that it is the number of labels/lines at line 1284.

So it seems to me that the correct fix is to: 

1.  Use `np.unique` to compute the set of labels.
2. Change the `for`-loop above to iterate over the set. 
3. Change conditions in line 1284 and this line to be that the set has size 1.

This is a bigger change. Let me know how you think!

Reply:
Ah you are correct, for the purpose of `name` we used K for the name implementation above, but that isn't actually the correct thing to be checking. I like your proposed solution!

Reply:
I updated the PR in new commit :)